frontend/src/components/User/UserPage.jsx

```diff
@@ -0,0 +1,227 @@
+      } else {
+        setError("Failed to fetch user details");
+      }
+    } catch (error) {
+      console.error("Error fetching user details:", error);
+      setError("Failed to fetch user details");
+    }
+  };
+
+  const handleLogout = () => {
+    logout();
+    navigate("/login");
+  };
+
+  const handleDeleteAccount = async () => {
+    try {
+      const response = await fetch(
+        `https://sunside-hotel.onrender.com/users/${userDetails?._id}`,
+        {
+          method: "DELETE",
+          headers: {
+            Authorization: accessToken,
+          },
+        }
+      );
+
+      if (response.ok) {
+        navigate("/login", {
+          state: {
+            successMessage: "Your account has been deleted successfully.",
+          },
+        });
+      } else {
+        setError("Failed to delete account");
+      }
+    } catch (error) {
+      console.error("Error deleting account:", error);
+      setError("Failed to delete account");
+    }
+  };
+
+  return (
+    <Container height={height} style={{ marginTop: "30px" }}>
+      <Content>
+        <Heading>BOOKING INFORMATION</Heading>
+        {bookings && bookings.length > 0 ? (
+         bookings.map((booking) => ( 
+          <UserInfoContainer key={booking.id}>
+            <div
+              style={{
+                display: "flex",
+                flexDirection: "row",
+                marginBottom: "10px",
+              }}
+            >
+              <UserDetails
+                style={{
+                  display: "flex",
+                  flexDirection: "column",
+                  alignItems: "center",
+                  marginRight: "15px",
+                }}
+              >
+                <StyledImage src={enter} alt="Check-in" />
+                <b>Check-in:</b>
+                <div>{new Date(booking.checkinDate).toLocaleDateString("sv-SE")}</div>
+              </UserDetails>
+              <UserDetails
+                style={{
+                  display: "flex",
+                  flexDirection: "column",
+                  alignItems: "center",
+                  marginRight: "15px",
+                }}
+              >
+                <StyledImage src={exit} alt="Check-out" />
+                <b>Check-out:</b>
+                <div>{new Date(booking.checkoutDate).toLocaleDateString("sv-SE")}</div>
+              </UserDetails>
+              <UserDetails
+                style={{
+                  display: "flex",
+                  flexDirection: "column",
+                  alignItems: "center",
+                }}
+              >
+                <StyledImage src={guest} alt="Guests" />
+                <b>Guests:</b>
+                <div>{booking.guests}</div>
+              </UserDetails>
+            </div>
+            <div style={{ textAlign: "center", marginTop: "10px" }}>
+              <BlackStyledLink to="/active-booking">
+                Click here to cancel your booking
+              </BlackStyledLink>
```

Comment:
what happens after cancelling the booking? There should be an info message, and if there's no other active bookings there should be an empty state component displayed 👍

Reply:
Now this function works on both frontend and backend. On backend, the booking will be deleted and room status will be changed back to be available. On frontend, there is an info message which shows that the booking is cancelled.